libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -582,6 +578,88 @@ public long getLastRefreshTime() {
 
         @Override
         public String getInstanceUrl() { return lastNewInstanceUrl; }
+
+        private Bundle refreshStaleToken(Account account) throws NetworkErrorException {
```

Comment:
A simple refresh module -> uses `OAuth2.java` to refresh the token and constructs a `Bundle` with the new access token and instance URL after updating data in `AccountManager`. Encryption of data is done to keep it interoperable with `AuthenticatorService`.